pages/product_grid/product_grid.js

```diff
@@ -4,12 +4,12 @@ import { useEffect } from 'react';
 import SkeletonCard from './skeleton_card';
 
 const fetcher = (...args) => fetch(...args).then((res) => res.json());
-const getKey = (pageIndex, previousPageData) => {
-  if (previousPageData && !previousPageData.length) return null; // reached the end
-  return `/api/products?orderReq=random&categoryReq=Shop_All&pageIndex=${pageIndex}&pageSize=20`; // SWR key
-};
 
-export default function ProductGrid({ inView }) {
+export default function ProductGrid({ inView, filter, sort }) {
+  const getKey = (pageIndex, previousPageData) => {
+    if (previousPageData && !previousPageData.length) return null; // reached the end
+    return `/api/products?orderReq=${sort}&categoryReq=${filter}&pageIndex=${pageIndex}&pageSize=20`; // SWR key
+  };
```

Comment:
I would opt to have this be a function that accepts additional params `filter` (which could be renamed `categoryFilter`) instead of it being inside of the component.

Reply:
Same for sort

Reply:
I changed the variable names, but I'm unable to add them as arguments to getKey as it messes up the function calls made by useSWRInfinite(getKey, fetcher)

Reply:
Ah that makes sense. And theres no way to overload the function, I think this is fine then